### PR TITLE
[Coastal area scenario] Add output class with basic render capabilities

### PR DESCRIPTION
**Pull request by fabiocruz**
This PR adds a class to represent outputs of the coastal area simulation scenario.

It can be used as other output classes to create renders, e.g.
```python

scenario = CoastalArea()
output = scenario.simulate()
output.render()
```

In this case, the `render` method creates movies of the temporal evolution of quantities computed by SWASH.

Here is the default output:

https://github.com/inductiva/inductiva/assets/11545632/3b1ac579-04b4-491c-acb9-09ba15eb2795

The rendered quantity can be configured by setting e.g. `quantity="velocity_magnitude"` in the call to `render`. The output in this case is the following:

https://github.com/inductiva/inductiva/assets/11545632/7c96caff-5ffd-4346-923a-734b114a30f2

### CODE REVIEW

**Review by IvanPombo — APPROVED** [review]
In general LGTM, I think adding a bit more description to the args would be perfect. 

It seems though all of the functions before the class can be used for all SWASH simulations, so I am not sure if we should keep them here or create a swash_viz_utils.py somewhere to be useful for low-level users as well.

File: inductiva/fluids/scenarios/coastal_area/output.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+    # The list of times is built from the keys of the dictionary:
+    # the keys end with a number, in format 123456_789, identifying the time
+    # instant 12 hours, 34 minutes, 56 seconds and 789 milliseconds; we
+    # strip the times from the keys and convert them to seconds.
+
+    # The list of data arrays is build from the values of the dictionary:
+    # the values correspond to the data values at each of the time instants
+    # represented by the corresponding keys; we read each value as a 2D
+    # tensor with indices (x, y).
```
Comment (cristianacarpinteiro) [inline inductiva/fluids/scenarios/coastal_area/output.py:95]: why not put this in the description of the function ?

File: inductiva/fluids/scenarios/coastal_area/output.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+            # again for consistency
+            transposed_data_values = data_values.transpose()
+
+            time_list.append(time_s)
+            data_list.append(transposed_data_values)
+
+    return (x_array, y_array), time_list, data_list
+
+
+def _render_quantity_data(x_array: np.ndarray,
```
Comment (cristianacarpinteiro) [inline inductiva/fluids/scenarios/coastal_area/output.py:123]: why not include these functions as methods of the class?

File: inductiva/fluids/scenarios/coastal_area/output.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+            "velocity_x",
+            "velocity_y",
+            "velocity_magnitude",
+        ] = "water_level",
+        movie_path: Path = "movie.mp4",
+        fps: int = 10,
+        cmap: str = "viridis",
+        clim: Optional[Sequence[float]] = None,
+    ):
+        """Renders temporal evolution of a quantity.
```
Comment (cristianacarpinteiro) [inline inductiva/fluids/scenarios/coastal_area/output.py:208]: maybe say what type of quantity, like a physical quantity such as velocity. Quantity seems too general!

**Review by cristianacarpinteiro — APPROVED** [review]
LGTM! Just left some comments to understand better your implementation!

File: inductiva/fluids/scenarios/coastal_area/output.py (hunk 2 above)
Comment (fabiocruz) [reply inductiva/fluids/scenarios/coastal_area/output.py:123]: As @IvanPombo said, part of these functions can be used to plot results of other SWASH simulations, so I didn't want to tie them to the class introduced here.

Comment (cristianacarpinteiro) [reply inductiva/fluids/scenarios/coastal_area/output.py]: Ah makes sense, then shouldn't it be elsewhere?

**Comment by fabiocruz** [comment]
> It seems though all of the functions before the class can be used for all SWASH simulations, so I am not sure if we should keep them here or create a swash_viz_utils.py somewhere to be useful for low-level users as well.

Sort of. These can be used to read and plot outputs of SWASH **defined over a grid**. SWASH exports these and other types of diagnostics, so it can't be used to plot arbitrary output files from SWASH. I added this specification to the names of the methods and corresponding docstrings.

This being said, I agree with you that these functions could be store somewhere where they could be used by other scenarios simulated with SWASH, or even by users using the low-level call. I don't have a good suggestion of where to put these now, though. Let's discuss this when we organize our visualization routines (soon, hopefully!), WDYT?

File: inductiva/fluids/scenarios/coastal_area/output.py (hunk 2 above)
Comment (fabiocruz) [reply inductiva/fluids/scenarios/coastal_area/output.py]: I think so, but I don't think this is the moment to make a decision about that. See [my comment](https://github.com/inductiva/inductiva/pull/396#issuecomment-1651421002) below.

**Comment by IvanPombo** [comment]
> > It seems though all of the functions before the class can be used for all SWASH simulations, so I am not sure if we should keep them here or create a swash_viz_utils.py somewhere to be useful for low-level users as well.
> 
> Sort of. These can be used to read and plot outputs of SWASH **defined over a grid**. SWASH exports these and other types of diagnostics, so it can't be used to plot arbitrary output files from SWASH. I added this specification to the names of the methods and corresponding docstrings.
> 
> This being said, I agree with you that these functions could be store somewhere where they could be used by other scenarios simulated with SWASH, or even by users using the low-level call. I don't have a good suggestion of where to put these now, though. Let's discuss this when we organize our visualization routines (soon, hopefully!), WDYT?

Copy!
Yes, it seems appropriate! We can move it to the right place at that point :)

**Pull request merged**